vm/src/vm/mod.rs

```diff
@@ -325,9 +334,21 @@ impl VirtualMachine {
         }
 
         #[cfg(feature = "encodings")]
-        if let Err(e) = self.import_encodings() {
-            eprintln!("encodings initialization failed. Only utf-8 encoding will be supported.");
-            self.print_exception(e);
+        if cfg!(feature = "freeze-stdlib") || !self.state.settings.path_list.is_empty() {
+            if let Err(e) = self.import_encodings() {
+                eprintln!(
+                    "encodings initialization failed. Only utf-8 encoding will be supported."
+                );
+                self.print_exception(e);
+            }
+        } else {
+            // Here may not be the best place to give general `path_list` advice,
+            // but bare rustpython_vm::VirtualMachine users skipped proper settings must hit here while properly setup vm never enters here.
+            eprintln!(
+                "feature `encodings` is enabled but `settings.path_list` is empty. This is likely a user-side bug. \
+                Please add library path to `settings.path_list`. If you intended to disable stdlib including `encodings`, please also disable `encodings` feature.\n\
+                Tip: You may also want to add `\"\"` to `settings.path_list` to enable importing from current working directory."
```

Comment:
@fanninpm I am sorry to add a commit during review. I found silently skipping this branch can lead users be confused. So this fix will be good to put here. Could you also check this message too?

Reply:
In my opinion, telling the user about "user error" seems a little bit off-putting. Other than that, I have a couple of minor changes:

```suggestion
            // Here may not be the best place to give general `path_list` advice,
            // but bare rustpython_vm::VirtualMachine users skipped proper settings must hit here while properly setup vm never enters here.
            eprintln!(
                "feature `encodings` is enabled but `settings.path_list` is empty. \
                Please add the library path to `settings.path_list`. If you intended to disable the entire standard library (including the `encodings` feature), please also make sure to disable the `encodings` feature.\n\
                Tip: You may also want to add `\"\"` to `settings.path_list` in order to enable importing from the current working directory."
```